pkgs/development/libraries/libvirt/0003-substitute-modules-path.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/src/util/virpci.c b/src/util/virpci.c
+index 780b4f9eec..0edb848d14 100644
+--- a/src/util/virpci.c
++++ b/src/util/virpci.c
+@@ -1461,7 +1461,9 @@ virPCIDeviceFindBestVFIOVariant(virPCIDevice *dev,
+     uname(&unameInfo);
+     modulesAliasPath = g_strdup_printf("/lib/modules/%s/modules.alias", unameInfo.release);
+     if (virFileReadAll(modulesAliasPath, 8 * 1024 * 1024, &modulesAliasContent) < 0)
+-        return -1;
++        modulesAliasPath = g_strdup_printf("/run/booted-system/kernel-modules/lib/modules/%s/modules.alias", unameInfo.release);
```

Comment:
You should `g_clear_pointer(&modulesAliasPath)`  before re-setting the variable or the original value will leak.

Reply:
So, the modulesAliasPath is already set up with g_autofree earlier in code. So, I don't think that g_clear_pointer(&modulesAliasPath) is needed; though I'm not too familiar with C. I did adjust the logic a bit though.

Reply:
`g_autofree` clears the value of the pointer at the end of the block scope where the variable is defined . If you overwrite the value of the variable before then, only the final value will be cleaned – unlike Rust or C++ RAII, C has no way to access the previous pointer values to free.

Reply:
Can you provide an example that I could update the PR with? I don't have enough experience with C here and this is going beyond what I have time to contribute at the moment.

Reply:
Just put `g_clear_pointer(&modulesAliasPath);` before line 12?

I don't have experience with gobject specifically, but what @jtojnar said should work.

Reply:
Having just `g_clear_pointer(&modulesAliasPath);` does not work since it requires 2 arguments. I've updated it to include `g_free` as the second argument. Does this look good now?

Reply:
@jtojnar

Reply:
Patch looks okay to me now.